pkg/build/api/types.go

```diff
@@ -328,6 +328,9 @@ type GitBuildSource struct {
 
 	// HTTPSProxy is a proxy used to reach the git repository over https
 	HTTPSProxy *string
+
+	// URLCheckTimeoutSeconds is the timeout in seconds to use when checking connectivity to the git server
```

Comment:
Add some detail about the default behavior and "normal" boundaries like nil, 0, -1.  Is the field defaulted or is it left empty (nil?, -1?) to indicate intent?  Is there a way to say, "never give up"?